internal/client.go

```diff
@@ -578,6 +581,9 @@ func NewClient(service workflowserviceclient.Interface, domain string, options *
 	if options != nil && options.Authorization != nil {
 		service = auth.NewWorkflowServiceWrapper(service, options.Authorization)
 	}
+	if options.IsolationGroup != "" {
```

Comment:
huh.  I'd say keep allowing nil, since it's fine if it's empty... but I think this is entirely internal, so we _could_ start requiring it 🤔.  we'd just have to pass an empty value when null in the public API, which is easy.

if we're going to start requiring it though, there's a lot of other `options != nil` checks that can be removed, since they'll just panic here if they aren't true.

Reply:
Not entirely sure I follow, I don't feel strongly about this check though, but I'm pretty sure we'll never require this field.  It's unlikely most clients will care about this feature.

Reply:
all other lines here include
```go
if options != nil && options.Authorization != nil {
   ^ this nil check
```
which allows options to be nil.

this one doesn't, so either all those != nil checks are unnecessary because we always pass options, or it's changing the function to require options all the time.

so I'd add the nil check to this one too.  or clean up if it actually is always passed.

Reply:
oh 🤦  sorry, good catch